src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/domainevents/DomainEventDescriber.kt

```diff
@@ -113,7 +112,11 @@ class DomainEventDescriber(
 
   private fun buildBookingNotMadeDescription(domainEventSummary: DomainEventSummary): String? {
     val event = domainEventService.getBookingNotMadeEvent(domainEventSummary.id())
-    return event.describe { "A placement was not made for the placement request. The reason was: ${it.eventDetails.failureDescription}" }
+    var failureReason = ""
+    if (event?.data?.eventDetails?.failureDescription != null) {
+      failureReason = " The reason was: ${event?.data?.eventDetails?.failureDescription}"
+    }
```

Comment:
In kotlin if statements can return values. I think we can use that along with the 'let' scope function and the elvis operator to remove the variable mutation whilst (hopefully) keeping it readible:

```
val failureReason = event?.data?.eventDetails?.failureDescription?.let { 
   " The reason was: $it" 
} ?: ""
```